contracts/Auctioneer.sol

```diff
@@ -62,6 +62,9 @@ contract Auctioneer is CloneFactory {
     ) external {
         require(openAuctions[msg.sender], "Sender isn't an auction");
 
+        // Check the state of the deposit associated with the auction
+        checkDepositState(msg.sender);
```

Comment:
`Auctioneer` should be abstract enough so it can be used by any risk manager implementation, also if that implementation does not have a concept of a deposit. I think we need function that is more generic, something that does not imply this contract works with tBTC.

Another note is that for `offerTaken` calls for fully filled auctions, [we already have the desired behavior](https://github.com/keep-network/tbtc/blob/762b189bdb1ce7b453aa5a4cd710e2092fdf378a/solidity/contracts/deposit/DepositLiquidation.sol#L140).

If instead of `checkDepositState` we do `onAuctionPartiallyFilled` next to `onAuctionFullyFilled`, we can have a more generic API and we can execute the deposit state check only for the latter given that it is already happening implicitly for the latter.

Reply:
I'd also feel safer if we had those two scenarios covered later in a system test with real tBTC contract. But #53 needs to be merged first.

Reply:
> If instead of checkDepositState we do onAuctionPartiallyFilled next to onAuctionFullyFilled, we can have a more generic API and we can execute the deposit state check only for the latter given that it is already happening implicitly for the latter.

I added function `onAuctionPartiallyFilled` and overridden it in `RiskManagerV1`:
```
function onAuctionPartiallyFilled(Auction auction) internal view override {
    IDeposit deposit = IDeposit(auctionToDeposit[address(auction)]);
    // Make sure the deposit was not liquidated outside of Coverage Pool
    require(
        deposit.currentState() == DEPOSIT_LIQUIDATION_IN_PROGRESS_STATE,
        "Deposit liquidation is not in progress"
    );
}
```

And the condition at the end of `offerTaken` looks like this:
```        
if (auction.isOpen()) {
    onAuctionPartiallyFilled(auction);
} else {
    onAuctionFullyFilled(auction);

    emit AuctionClosed(msg.sender);
    delete openAuctions[msg.sender];
}
```

Reply:
> I'd also feel safer if we had those two scenarios covered later in a system test with real tBTC contract. But #53 needs to be merged first.

But should it be done in a separate PR?
Should we add a task to the backlog?

Reply:
Yes, let's backlog it and work on it in a separate PR, no reason to block this one.